src/assets.zig

```diff
@@ -402,30 +456,46 @@ pub fn loadWorldAssets(assetFolder: []const u8, blockPalette: *Palette, biomePal
 		_ = main.models.registerModel(entry.key_ptr.*, entry.value_ptr.*);
 	}
 
+	// items #1
+	for(itemPalette.palette.items) |id| {
+		// Normal items should appear in items hash map.
+		if(items.get(id)) |zon| {
+			try registerItem(assetFolder, id, zon);
+			continue;
+		}
+		// But there are also items created automatically from blocks.
+		if(blocks.get(id)) |zon| {
+			if(!zon.get(bool, "hasItem", true)) continue;
+			try registerItem(assetFolder, id, zon.getChild("item"));
```

Comment:
Shouldn't this be calling assignBlockItem?

Reply:
No, assignment is done after all items were registered from palette, from common items and from block palette and from block items. This ensures we don't miss any items (or block-items).

Reply:
This code is so confusing, fr

Reply:
I think no. Here we are only registering the items. We are assigning them in separate loop after we are sure we have complete collection of them.